hbogolib/handlereu.py

```diff
@@ -1185,20 +1194,41 @@ def procContext(self, action_type, content_id, optional=""):
 
 
     def genContextMenu(self, content_id, media_id):
+        runplugin = 'RunPlugin(%s?%s)'
 
-        add_mylist = (py2_encode(self.language(30719)), 'RunPlugin(' + self.base_url + "?url=ADDMYLIST&mode=9&cid=" + media_id + ')')
-        remove_mylist = (py2_encode(self.language(30720)), 'RunPlugin(' + self.base_url + "?url=REMMYLIST&mode=10&cid=" + media_id + ')')
+        add_mylist_query = urlencode({
+            'url': 'ADDMYLIST',
+            'mode': 9,
+            'cid': media_id,
+            })
+        add_mylist = (py2_encode(self.language(30719)), runplugin % (self.base_url, add_mylist_query))
 
-        vote_5 = (py2_encode(self.language(30721)), 'RunPlugin(' + self.base_url + "?url=VOTE&mode=8&vote=5&cid=" + content_id + ')')
-        vote_4 = (py2_encode(self.language(30722)), 'RunPlugin(' + self.base_url + "?url=VOTE&mode=8&vote=4&cid=" + content_id + ')')
-        vote_3 = (py2_encode(self.language(30723)), 'RunPlugin(' + self.base_url + "?url=VOTE&mode=8&vote=3&cid=" + content_id + ')')
-        vote_2 = (py2_encode(self.language(30724)), 'RunPlugin(' + self.base_url + "?url=VOTE&mode=8&vote=2&cid=" + content_id + ')')
-        vote_1 = (py2_encode(self.language(30725)), 'RunPlugin(' + self.base_url + "?url=VOTE&mode=8&vote=1&cid=" + content_id + ')')
+        remove_mylist_query = urlencode({
+            'url': 'REMMYLIST',
+            'mode': 10,
+            'cid': media_id,
+            })
+        remove_mylist = (py2_encode(self.language(30720)), runplugin % (self.base_url, remove_mylist_query))
+
+        votes_configs = [
+            {'str_id': 30721, 'vote': 5},
+            {'str_id': 30722, 'vote': 4},
+            {'str_id': 30723, 'vote': 3},
+            {'str_id': 30724, 'vote': 2},
+            {'str_id': 30725, 'vote': 1},
+        ]
+
+        votes = map(lambda item: (py2_encode(self.language(item['str_id'])), runplugin % (self.base_url, urlencode({
+            'url': 'VOTE',
+            'mode': 8,
+            'vote': item['vote'],
+            'cid': content_id,
+            }))), votes_configs)
 
         if self.cur_loc == self.LB_MYPLAYLIST:
-            return [vote_5, vote_4, vote_3, vote_2, vote_1, remove_mylist]
+            return votes + [remove_mylist]
```

Comment:
This fails in Python 3, didn't have time to look into details:

```
ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <class 'TypeError'>
                                            Error Contents: unsupported operand type(s) for +: 'map' and 'list'
                                            Traceback (most recent call last):
                                              File "/Users/luka/Library/Application Support/Kodi/addons/plugin.video.hbogoeu/main.py", line 26, in <module>
                                                addon_main.router(REQUEST_PARAMS)
                                              File "/Users/luka/Library/Application Support/Kodi/addons/plugin.video.hbogoeu/hbogolib/base.py", line 140, in router
                                                self.handler.list(url)
                                              File "/Users/luka/Library/Application Support/Kodi/addons/plugin.video.hbogoeu/hbogolib/handlereu.py", line 911, in list
                                                self.addDir(title, 2, "tvshow")
                                              File "/Users/luka/Library/Application Support/Kodi/addons/plugin.video.hbogoeu/hbogolib/handlereu.py", line 1335, in addDir
                                                liz.addContextMenuItems(items=self.genContextMenu(cid, media_id))
                                              File "/Users/luka/Library/Application Support/Kodi/addons/plugin.video.hbogoeu/hbogolib/handlereu.py", line 1229, in genContextMenu
                                                return votes + [remove_mylist]
                                            TypeError: unsupported operand type(s) for +: 'map' and 'list'
                                        
```

Reply:
Fixed it